flask_restless/views/base.py

```diff
@@ -1617,7 +1617,7 @@ def _get_collection_helper(self, resource=None, relation_name=None,
                 result['data'] = serialize(data, only=only)
             except SerializationException as exception:
                 return errors_from_serialization_exceptions([exception])
-            primary_key = self.primary_key or primary_key_name(data)
+            primary_key = self.primary_key or primary_key_for(data)
```

Comment:
The `self.primary_key or` part is no longer needed, since the APIManager informs the `primary_key_for` function what the primary key is.

(In fact, the `self.primary_key` is not really needed at all anymore; with the introduction of `primary_key_for`, it is used only as an optimization to avoid the additional function call. That's a different issue...)